src/apps/relay/ns_ioalib_engine_impl.c

```diff
@@ -3596,6 +3596,14 @@ void turn_report_allocation_delete(void *a)
 					}
 					send_message_to_redis(e->rch, "del", key, "");
 					send_message_to_redis(e->rch, "publish", key, "deleted");
+
+					// report total traffic usage for this allocation
+					if(ss->realm_options.name[0]) {
+						snprintf(key, sizeof(key), "turn/realm/%s/user/%s/allocation/%018llu/total_traffic", ss->realm_options.name, (char*)ss->username, (unsigned long long)ss->id);
```

Comment:
isnt' "/" a valid character in the realm and username?  Shouldn't you escape it?

Reply:
True.
It is a general issue with the db backend.

Reply:
I hadn't considered that, actually. I just followed the pattern for the other Redis keys.

Reply:
I think I will merge this and open a new issue about the string escaping.

Reply:
@ggarber Many thanks again for pointing out the issue..